src/requestor/routes/query.py

```diff
@@ -99,29 +94,32 @@ async def list_user_requests(api_request: Request, auth=Depends(Auth)) -> dict:
     """
     List current user's requests.
 
-    Use "active" query parameter to get only those requests
+    Use the "active" query parameter to get only those requests
     created by the user that are not in DRAFT or FINAL statuses.
     """
     # no authz checks because we assume the current user can read
     # their own requests.
     active = False
-    filter_dict = {}
-    flat_params = flatten_query_params(api_request.query_params)
-    for param in flat_params:
+    filter_dict = {k: [] for k in api_request.query_params if k != "active"}
+    for param, value in api_request.query_params.multi_items():
         if param == "active":
-            if flat_params["active"]:
+            if value:
                 raise HTTPException(
                     HTTP_400_BAD_REQUEST,
-                    f"The 'active' parameter should not be assigned a value. Received '{flat_params['active']}'",
+                    f"The 'active' parameter should not be assigned a value. Received '{value}'",
                 )
             active = True
         elif not hasattr(RequestModel, param):
             raise HTTPException(
                 HTTP_400_BAD_REQUEST,
                 f"The '{param}' parameter is invalid.",
             )
-        else:
-            filter_dict[param] = flat_params[param]
+        elif value:
+            if str(getattr(RequestModel, param).type) == "BOOLEAN":
+                value = value == "True"
+            elif str(getattr(RequestModel, param).type) == "DATETIME":
```

Comment:
would [this](https://github.com/uc-cdis/audit-service/blob/ec29058/src/audit/routes/query.py#L220) work instead of comparing strings? just sharing the audit-service implementation because i've struggled with this problem before :p

Reply:
Woah! This is great, I didn't know this exists. Thanks :D